prover/protocol/distributed/distributed.go

```diff
@@ -0,0 +1,79 @@
+package distributed
+
+import (
+	"github.com/consensys/linea-monorepo/prover/protocol/ifaces"
+	"github.com/consensys/linea-monorepo/prover/protocol/wizard"
+)
+
+type moduleName = string
+
+type DistributedWizard struct {
+	// initializedWizard
+	Bootstrapper *wizard.CompiledIOP
+	// compiledIOPs for a segment of each module.
+	// Since splits are fair/uniform over a module. One segment can represent all the segments.
+	DistributedModules []DistributedModule
+	// it is a compiledIOP object representing the consistency checks among the segments.
+	Aggregator *wizard.CompiledIOP
+}
+
+// DistributedModule implements the utilities relevant to a single segment from a module.
+type DistributedModule struct {
+	LookupPermProj *wizard.CompiledIOP
+	GlobalLocal    *wizard.CompiledIOP
+}
+
+// ModuleDiscoverer a set of methods responsible for the horizontal splittings (i.e., splitting to modules)
+type ModuleDiscoverer interface {
+	// Analyze is responsible for letting the module discoverer compute how to
+	// group best the columns into modules.
+	Analyze(comp *wizard.CompiledIOP)
+	ModuleList(comp *wizard.CompiledIOP) []string
+	FindModule(col ifaces.Column) moduleName
+	FindNbSegments(moduleName string, SegParam map[string]int) int
```

Comment:
A comment describing SegParam is helpful.

Reply:
Why is that a member of the module discovery interface? And what are the parameters? From the code I understand that the function returns `Segparams[moduleName]`, you don't need a function to do that